### ISSUE TITLE
API Subdomain breaks docker images

### ISSUE THREAD

**Issue by encima** [gristlabs/grist-core#10]
Currently, the docker instructions do not match the latest version of the code (unless you have a host set up)

[gristUrls](https://github.com/gristlabs/grist-core/blob/a1a84d99c0e06ed2eb52b67cf56d808fa6d9dc9e/app/common/gristUrls.ts) has a default subdomain of 'api' which, when run with docker, will prepend `api` with the IP of the host.

i.e. 
![image](https://user-images.githubusercontent.com/517923/111898610-7cf94280-8a27-11eb-877d-00bcfdeefc54.png)

Could we use the `api` route by default instead and make this configurable through an environment variable, perhaps?

**Comment by paulfitz** [comment]
Thanks for the heads-up @encima. Just so I understand, do you have trouble if you visit `http://localhost:8484` after running this:
```
docker pull gristlabs/grist
docker run -p 8484:8484 -it gristlabs/grist
```

If you know the IP address that grist will ultimately be served on, you can set some environment variables to let it know. Would this work for you?:

```
export IP=192.168.1.153
docker run -p 8484:8484 \
       --env GRIST_SINGLE_PORT=true \
       --env GRIST_HOST=0.0.0.0 \
       --env GRIST_SINGLE_ORG=docs \
       --env GRIST_DOMAIN=$IP \
       --env APP_HOME_URL=http://$IP:8484 \
       --env APP_DOC_URL=http://$IP:8484 \
       -it gristlabs/grist
```

I agree it'd be better if this worked straight out of the box, without the `api` logic confounding things. We also also documentation of environment variables we should make public...

**Comment by encima** [comment]
I did find the environment variables so that part was ok. but just running with local host causes an issue because connection string tries api.localhost and fails. 

also, when I set the IP under APP_HOME etc then the api will then try api.168.1.2 instead of api.$IP. Both would fail but I guess the regex is just splitting at the first "."?

**Comment by paulfitz** [comment]
Would it be possible to describe exactly how you are running Grist?  Then I could help suggest environment variable changes that might help you.  Variables that may be relevant include `GRIST_SINGLE_PORT`, `GRIST_SINGLE_ORG`, and `GRIST_ORG_IN_PATH`.  Accesses to `api.*` should be fixable with a configuration matching the situation (running a single server handling all responsibilities, and perhaps for a single hard-coded organization).

**Comment by encima** [comment]
Sorry for the delay here! I did switch to paying for an account with your hosted version but would still like a mirrored, self hosted version.

Currently, I am deploying all in a single container with a Lets Encrypt proxy pointing to the grist domain

I tested running on my local machine and all works fine so I assume it is something with the proxy (or Caprovers port mapping); I will close now and update if I do succeed in running it this way.
